pkg/kubelet/kuberuntime/kuberuntime_container.go

```diff
@@ -185,6 +185,37 @@ func makeMounts(cid string, opts *kubecontainer.RunContainerOptions, container *
 	return volumeMounts
 }
 
+// getKubeletContainers lists all (or just the running) containers managed by kubelet.
+func (m *kubeGenericRuntimeManager) getKubeletContainers(allContainers bool) ([]*runtimeApi.Container, error) {
+	var resp []*runtimeApi.Container
+	var err error
+
+	if allContainers {
+		resp, err = m.runtimeService.ListContainers(nil)
+	} else {
+		runningState := runtimeApi.ContainerState_RUNNING
+		resp, err = m.runtimeService.ListContainers(&runtimeApi.ContainerFilter{
+			State: &runningState,
+		})
+	}
+	if err != nil {
+		glog.Errorf("ListContainers failed: %v", err)
+		return nil, err
+	}
+
+	result := []*runtimeApi.Container{}
+	for _, c := range resp {
+		if !isContainerManagedByKubelet(c.GetName()) {
```

Comment:
I think this goes back to the question of whether the runtime CRI implementation is used by kubelet exclusively.  Also, if kubelet manages this, it'd have the ability to handle versioning itself when needed. For example, handling older labels or names, without relying on the runtime. 

/cc @dchen1107